hack/test.sh

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env bash
+#
+# Test the OpenLDAP image.
+#
+# IMAGE specifies the name of the candidate image used for testing.
+# The image has to be available before this script is executed.
+#
+
+set -eo nounset
+shopt -s nullglob
+
+IMAGE=${IMAGE:-openshift/openldap-candidate}
+RUNTIME=${RUNTIME:-podman}
+
+CIDFILE_DIR=$(mktemp --suffix=openldap_test_cidfiles -d)
+
+function cleanup() {
+  for cidfile in $CIDFILE_DIR/* ; do
+    CONTAINER=$(cat $cidfile)
+    echo "Stopping and removing container $CONTAINER..."
+
+    $RUNTIME stop $CONTAINER
+    exit_status=$($RUNTIME inspect -f '{{.State.ExitCode}}' $CONTAINER)
+    if [ "$exit_status" != "0" ]; then
+      echo "Dumping logs for $CONTAINER"
+      $RUNTIME logs $CONTAINER
+    fi
+    $RUNTIME rm $CONTAINER
+    rm $cidfile
+    echo "Done."
+  done
+
+  rmdir $CIDFILE_DIR
+}
+
+function get_cid() {
+  local id="$1" ; shift || return 1
+
+  cat "$CIDFILE_DIR/$id"
+}
+
+function test_connection() {
+  local name=$1
+  local port=$2
+  echo "  Testing OpenLDAP connection to localhost:$port..."
+
+  local max_attempts=20
+  local sleep_time=2
+  for _ in $(seq $max_attempts); do
+    echo "    Trying to connect..."
+    set +e
+    ldapsearch -x -h localhost -p $port -b dc=example,dc=com objectClass=*
+    status=$?
+    set -e
+    if [ $status -eq 0 ]; then
+      echo "  Success!"
+      return 0
+    fi
+    sleep $sleep_time
+  done
+  echo "  Giving up: Failed to connect. Logs:"
+  $RUNTIME logs "$(get_cid $name)"
+
+  return 1
+}
+
+function test_openldap() {
+  local port=$1
+  echo "  Testing OpenLDAP"
+
+  ldapsearch -x -LLL -h localhost -p $port -b dc=example,dc=com objectClass=organization | grep "dc=example,dc=com"
```

Comment:
@stlaz, @s-urbaniak: I would assume, that we need to move that into a container right?

A cheap shot would be to `s/ldapsearch/$(RUNTIME) exec --rm bitnami/openldap ldapsearch`.
Otherwise I would need to:

- manage the bitnami/openldap container OR
- use the created image to manage another container.